Python/suggestions.c

```diff
@@ -250,6 +271,10 @@ get_suggestions_for_name_error(PyObject* name, PyFrameObject* frame)
     Py_DECREF(dir);
 
     return suggestions;
+
+error:
+    Py_XDECREF(dir);
```

Comment:
If I am not missing any, all the `goto` sites seem to have `dir` as a `non-NULL` value.
```suggestion
    Py_DECREF(dir);
```